packages/account/src/Components/additional-kyc-info-modal/additional-kyc-info-modal.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { observer, useStore } from '@deriv/stores';
+import {
+    Modal,
+    DesktopWrapper,
+    MobileWrapper,
+    UILoader,
+    PageOverlay,
+    Div100vhContainer,
+    HintBox,
+    Text,
+} from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import AdditionalKycInfoForm from './additional-kyc-info-form';
+
+type TAdditionalKycInfoFormWithHintBox = {
+    error: string;
+    setError: React.Dispatch<React.SetStateAction<string>>;
+};
+
+const AdditionalKycInfoFormWithHintBox = ({ error, setError }: TAdditionalKycInfoFormWithHintBox) => {
+    return (
+        <React.Fragment>
+            {error && (
+                <HintBox
+                    className='additional-kyc-info-modal__hintbox'
+                    icon='IcAlertDanger'
+                    icon_height={16}
+                    icon_width={16}
+                    message={
+                        <Text as='p' size='xxxs'>
+                            {error}
+                        </Text>
+                    }
+                    is_danger
+                />
+            )}
+            <AdditionalKycInfoForm setError={setError} />
+        </React.Fragment>
+    );
+};
+
+export const AdditionalKycInfoModal = observer(() => {
+    const {
+        ui: { is_additional_kyc_info_modal_open: is_open, toggleAdditionalKycInfoModal },
+    } = useStore();
+    const [error, setError] = React.useState('');
+
+    const toggleModal = (e?: React.MouseEvent<HTMLElement, MouseEvent> | undefined) => {
+        // if e.target is anchor tag, don't close modal for link click within modal
+        const target = e?.target as HTMLElement;
+        if (target.tagName === 'A') e?.stopPropagation();
+        toggleAdditionalKycInfoModal();
+    };
+
+    return (
+        <React.Suspense fallback={<UILoader />}>
+            <div className='additional-kyc-info-modal__container'>
+                <DesktopWrapper>
+                    <Modal
+                        has_close_icon
+                        is_open={is_open}
+                        title={
+                            <Localize i18n_default_text='Additional information required for Deriv MT5 account(s)' />
+                        }
+                        toggleModal={toggleModal}
+                        className='additional-kyc-info'
+                        width='90.4rem'
+                        height={error ? '52.8rem' : '49.6rem'}
+                    >
+                        <Modal.Body className='additional-kyc-info-modal__form'>
+                            <AdditionalKycInfoFormWithHintBox setError={setError} error={error} />
+                        </Modal.Body>
+                    </Modal>
+                </DesktopWrapper>
+                <MobileWrapper>
```

Comment:
@shahzaib-deriv Could we not use these wrappers and simply create 2 different components for desktop and mobile and switch between them based on the UIStore value? 🙏🏻

Reply:
@shahzaib-deriv Yes if it's a small component I guess we don't need to break it down even further and can handle the condition in that component, Just a suggestion tho, Go with anything you feel is the best 👍🏻

Reply:
Thanks @farzin-deriv we've plan for refactoring the forms within accounts and we'll definitely create separate fields for each case with it's own field level validation as well

Reply:
Created a separate field for select component 👍